bezier-rs/lib/src/lib.rs

```diff
@@ -624,37 +640,43 @@ impl Bezier {
-	fn intersections_between_subcurves(&self, self_original_t_interval: [f64; 2], other: &Bezier, error: f64) -> Vec<f64> {
+	/// - `other_original_t_interval` - Used to identify the `t` values of the original parent of `other` that the current iteration is representing.
+	fn intersections_between_subcurves(&self, self_original_t_interval: [f64; 2], other: &Bezier, other_original_t_interval: [f64; 2], error: f64) -> Vec<[f64; 2]> {
 		let bounding_box1 = self.bounding_box();
 		let bounding_box2 = other.bounding_box();
 
 		// Get the `t` interval of the original parent of `self` and determine the middle `t` value
 		let [curve1_start_t, curve1_end_t] = self_original_t_interval;
 		let curve1_mid_t = curve1_start_t + (curve1_end_t - curve1_start_t) / 2.;
 
+		// Get the `t` interval of the original parent of `other` and determine the middle `t` value
+		let [curve2_start_t, curve2_end_t] = other_original_t_interval;
+		let curve2_mid_t = curve2_start_t + (curve2_end_t - curve2_start_t) / 2.;
+
 		let error_threshold = DVec2::new(error, error);
 
 		// Check if the bounding boxes overlap
 		if utils::do_rectangles_overlap(bounding_box1, bounding_box2) {
 			// If bounding boxes are within the error threshold (i.e. are small enough), we have found an intersection
 			if (bounding_box1[1] - bounding_box1[0]).lt(&error_threshold) && (bounding_box2[1] - bounding_box2[0]).lt(&error_threshold) {
-				// Use the middle t value
-				return vec![curve1_mid_t];
+				// Use the middle t value, return the corresponding `t` value for `self` and `other`
+				return vec![[curve1_mid_t, curve2_mid_t]];
 			}
 
 			// Split curves in half and repeat with the combinations of the two halves of each curve
 			let [split_1_a, split_1_b] = self.split(0.5);
 			let [split_2_a, split_2_b] = other.split(0.5);
 
-			// Get the new `t` intervals for the split halves of `self`
+			// Get the new `t` intervals for the split halves of `self` and `other`
 			let interval_1_a = [curve1_start_t, curve1_mid_t];
 			let interval_1_b = [curve1_mid_t, curve1_end_t];
+			let interval_2_a = [curve2_start_t, curve2_mid_t];
+			let interval_2_b = [curve2_mid_t, curve2_end_t];
 			[
-				split_1_a.intersections_between_subcurves(interval_1_a, &split_2_a, error),
-				split_1_a.intersections_between_subcurves(interval_1_a, &split_2_b, error),
-				split_1_b.intersections_between_subcurves(interval_1_b, &split_2_a, error),
-				split_1_b.intersections_between_subcurves(interval_1_b, &split_2_b, error),
+				split_1_a.intersections_between_subcurves(interval_1_a, &split_2_a, interval_2_a, error),
+				split_1_a.intersections_between_subcurves(interval_1_a, &split_2_b, interval_2_b, error),
+				split_1_b.intersections_between_subcurves(interval_1_b, &split_2_a, interval_2_a, error),
+				split_1_b.intersections_between_subcurves(interval_1_b, &split_2_b, interval_2_b, error),
 			]
 			.concat()
 		} else {
```

Comment:
A few lines below (GitHub doesn't let me comment on it directly), the `fn intersections` has a parameter named `curve`, can you rename that `other` please?